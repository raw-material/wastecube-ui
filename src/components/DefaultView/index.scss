.default-view {
    padding: 0 20px;

    figure {
        height: 85vh;
        width: 100%;
        margin: 0;
        display: flex;

        img {
            min-width: 100%;
            min-height: 100%;
            object-fit: contain;
        }
    }

    .button-wrapper {
        height: 15vh;
        display: flex;
        justify-content: center;
        align-items: center;

        button.start-button {
            font-size: 2rem;
            border: 0;
        }
    }
}
