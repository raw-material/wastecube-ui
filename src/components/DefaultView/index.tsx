import * as React from 'react';
import { IAd } from '../App';
import './index.scss';

interface DefaultStateProps {
  ads?: Array<IAd>;
  changeState: (action: number) => void;
}

const DefaultView: React.FC<DefaultStateProps> = ({
  ads = [],
  changeState
}) => (
  <div className="default-view">
    {ads.length > 0 && (
      <figure>
        <img alt={ads[0].title} src={`http://localhost:5000${ads[0].url}`} />
      </figure>
    )}
    <div className="button-wrapper">
      <button className="start-button" onClick={() => changeState(1)}>
        Tocca per iniziare
      </button>
    </div>
  </div>
);

export default DefaultView;
