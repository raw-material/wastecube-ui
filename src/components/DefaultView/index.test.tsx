import React from 'react';
import * as renderer from 'react-test-renderer';
import { cleanup, render } from '@testing-library/react';
import { axe, toHaveNoViolations } from 'jest-axe';
import DefaultView from './index';

expect.extend(toHaveNoViolations);
afterEach(cleanup);

it('renders default view', () => {
  const component = renderer.create(<DefaultView changeState={() => {}} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
