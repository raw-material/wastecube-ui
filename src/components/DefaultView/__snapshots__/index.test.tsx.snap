// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default view 1`] = `
<div
  className="default-view"
>
  <figure>
    <img
      alt="bottles and cans"
      src="bottles-and-cans.jpg"
    />
  </figure>
  <div
    className="button-wrapper"
  >
    <button
      className="start-button"
      onClick={[Function]}
    >
      Tocca per iniziare
    </button>
  </div>
</div>
`;
